calcite-qml/Calcite/SpinBox.qml

```diff
@@ -0,0 +1,146 @@
+                             down.implicitIndicatorHeight)
+    topPadding: 5
+    bottomPadding: 5
+
+    spacing: 6
+
+    leftPadding: (control.mirrored ? (up.indicator ? up.indicator.width : 0) : (down.indicator ? down.indicator.width : 0))
+    rightPadding: (control.mirrored ? (down.indicator ? down.indicator.width : 0) : (up.indicator ? up.indicator.width : 0))
+
+    validator: IntValidator {
+        locale: control.locale.name
+        bottom: Math.min(control.from, control.to)
+        top: Math.max(control.from, control.to)
+    }
+
+    contentItem: TextInput {
+        id: textInput
+        text: control.displayText
+
+        font: control.font
+        color : Calcite.text1
+        selectionColor: Calcite.brandHover
+        selectedTextColor: Calcite.foreground1
+        horizontalAlignment: Qt.AlignHCenter
+        verticalAlignment: Qt.AlignVCenter
+
+        readOnly: !control.editable
+        validator: control.validator
+        inputMethodHints: control.inputMethodHints
+    }
+
+    up.indicator: Item {
+        x: control.mirrored ? 0 : parent.width - width
+        implicitWidth: 32
+        implicitHeight: 32
+        height: parent.height
+        width: height
+        Rectangle {
+            //y: parent.y + parent.height - height - control.bottomPadding / 2
+            width: parent.width
+            height: parent.width
+            color: enabled ? (up.hovered ? Calcite.foreground2 : Calcite.foreground1) : Calcite.background
+            border {
+                color: Calcite.border1
+            }
+
+            Image {
+                id: upIndicatorImage
+                fillMode: Image.PreserveAspectFit
+                anchors {
+                    fill: parent
+                }
+                source: "images/chevron-right.svg"
+                sourceSize.width: parent.width
+                sourceSize.height: parent.width
+            }
+
+            ColorOverlay {
+                anchors.fill : upIndicatorImage
+                source: upIndicatorImage
+                color: up.hovered ? Calcite.text1 : Calcite.text3
+            }
+        }
+    }
+
+
+    down.indicator: Item {
+        x: control.mirrored ? parent.width - width : 0
+        implicitWidth: 32
+        implicitHeight: 32
+        height: parent.height
+        width: height
+
+        Rectangle {
+            width: parent.width
+            height: parent.height
+            color: enabled ? (down.hovered ? Calcite.foreground2 : Calcite.foreground1) : Calcite.background
+            border {
+                color: Calcite.border1
+            }
+
+            Image {
+                id: downIndicatorImage
+                fillMode: Image.PreserveAspectFit
+                anchors {
+                    fill: parent
+                }
+                source: "images/chevron-left.svg"
+                sourceSize.width: parent.width
+                sourceSize.height: parent.width
```

Comment:
I noticed a major difference in using the sourceSize to scale the `svg` image, compared to only using the `fill : parent`. Qt discourages to binding `sourceSize` for svg because it can be slow, but below is a screenshot of not using it (left arrow).
![image](https://user-images.githubusercontent.com/91129623/148267534-57e82c8c-f0f6-444c-91cb-1a7bcea8b383.png)

Reply:
Taking a look at the [Image.sourceSize](https://doc.qt.io/qt-5/qml-qtquick-image.html#sourceSize-prop) docs:

>  If the source is an intrinsically scalable image (eg. SVG), this property determines the size of the loaded image regardless of intrinsic size. Avoid changing this property dynamically; rendering an SVG is slow compared to an image.

I would interpret "dynamically" to mean binding sourceSize to a changing value. Here, you are binding to a value that does not change for the lifetime of the SpinBox, so I don't think the performance problems will be hit.

❓ Should it be `sourceSize.height: parent.width` or `sourceSize.height: parent.height`?

Reply:
should be `height`, you are correct